lib/languageFeatures/findReferencesHandler.ts

```diff
@@ -56,16 +56,18 @@ export async function prepareRenameHandler(params: TextDocumentPositionParams) {
   if (!candidate) {
     throw new ResponseError(ErrorCodes.InvalidRequest, 'Can not rename this element', 'Can not rename this element');
   }
+  if (implementsIHasLexerToken(candidate)) {
+    return candidate.lexerToken.range;
+  }
   return candidate.range;
 }
 export async function renameHandler(params: RenameParams) {
   const references = (await findReferences(params)).map(reference => {
-    if (implementsIReferencable(reference) && implementsIHasLexerToken(reference)) {
+    if (implementsIHasLexerToken(reference)) {
       return reference.lexerToken;
     }
     return reference;
-  });
-  // console.log(references.map(reference => `${reference.range.start.line} ${reference.range.start.character}`));
+  }).filter((r, i, s) => s.indexOf(r) === i);
```

Comment:
What is this? Explain maybe

Reply:
Ah, the parameters are default filter, then it is always true oO?
Ah it picks only the first If the exact same is in it twice? why not use Set?

Reply:
I made it clearer